File: api/apis/builds.py
from flask_restx import Namespace, Resource
from flask import request

from repo.builds import add_build_and_gen_insights
from utils.response import Response

api = Namespace('builds', description='Build related operations')


@api.route('/')
class BuildOperations(Resource):
    def post(self):
        payload = request.json
        build_id = add_build_and_gen_insights(payload)
        response = Response(status=201, data=build_id)
        return response.success(), response.status

File: client/src/apis/http/endpoints/builds.ts
import { IBuild } from "@store/build/types";
import { api } from "..";

const GET_ONE_BUILD_URL = (buildId: string) => `/builds/${buildId}`;

export const getBuildData = async (buildId: string) =>
  await api.get<IBuild>(GET_ONE_BUILD_URL(buildId));
